packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -2442,6 +2444,31 @@ export class MergeTree {
         }
     }
 
+    public removeLocalReferencePosition(lref: ReferencePosition): ReferencePosition | undefined {
+        const segment = lref.getSegment();
+        if (segment) {
+            const removedRefs = segment?.localRefs?.removeLocalRef(lref);
+            if (removedRefs !== undefined) {
+                this.blockUpdatePathLengths(segment.parent, TreeMaintenanceSequenceNumber,
+                    LocalClientId);
+            }
+            return removedRefs;
+        }
+    }
+    public createLocalReferencePosition(
+        segment: ISegment, offset: number, refType: ReferenceType, properties: PropertySet | undefined,
+        client: Client,
+    ): ReferencePosition {
+        const localRefs = segment.localRefs ?? new LocalReferenceCollection(segment);
+        segment.localRefs = localRefs;
+
+        const segRef = localRefs.createLocalRef(offset, refType, properties, client);
+
+        this.blockUpdatePathLengths(segment.parent, TreeMaintenanceSequenceNumber,
```

Comment:
Why are the calls to blockUpdatePathLengths required? I wouldn't expect creating/removing a reference to change the path lengths.

Reply:
looks to be for bookkeeping on local references in `HierBlock`s, but yeah naming seems confusing

Reply:
the handling for markers vs local references is very confused in the code. i haven't had time to tease it all apart, so for now just going for a do no harm approach where we follow the existing patterns.